src/tokio_util.rs

```diff
@@ -402,3 +402,27 @@ pub(crate) async fn read_as_bytes(reader: &mut Either<Cursor<Bytes>, File>) -> i
         }
     }
 }
+
+/// A join handle that owns the task it is running, and aborts it when dropped.
+#[derive(Debug)]
+pub(crate) struct OwnedJoinHandle<T>(tokio::task::JoinHandle<T>);
```

Comment:
bikeshedding: `OwningJoinHandle`.  The handle owns the task, the handle itself is not owned by something.

Though I'd be tempted to even call this `AbortingJoinHandle`